src/app/users/postConfirmDeactivate.js

```diff
@@ -17,8 +19,18 @@ const updateUserIndex = async (uid, correlationId) => {
 const postConfirmDeactivate = async (req, res) => {
   const user = await getUserDetails(req);
 
-  await deactivate(user.id, req.id);
-  await updateUserIndex(user.id, req.id);
+  if (req.body.reason === '') {
```

Comment:
I would add here a regex to prevent user to send a whitespace string - something like this 
`if (req.body.reason.match(/^\s*$/) !== null) {`